TM1py/Services/RestService.py

```diff
@@ -786,8 +786,22 @@ def _start_session(self, user: str, password: str, decode_b64: bool = False, nam
         finally:
             # If the TM1 REST API is routed through a reverse proxy that alters the expected URL,
             # we explicitly re-set the 'TM1SessionId' cookie to maintain session continuity.
-            session_id = self._s.cookies.pop('TM1SessionId', None)
-            if session_id is not None:
+            session_id = None
+            for cookie in self._s.cookies:
+                if cookie.name == 'TM1SessionId':
+                    session_id = cookie.value
+                    # break  # Use the first match
```

Comment:
Maybe this is an obsolete comment or did you want to break after the first match?